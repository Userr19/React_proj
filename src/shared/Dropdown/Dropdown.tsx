import React, { useEffect, useState } from 'react';
import styles from './dropdown.css';

interface IDropdownProps {
  button: React.ReactNode,
  karma: React.ReactNode,
  children: React.ReactNode,
  isOpen: boolean,
  onOpen?: () => void,
  onClose?: () => void,
}

const NOOP = () => {

}

export function Dropdown({ button, karma, children, isOpen = false, onClose = NOOP, onOpen = NOOP }: IDropdownProps) {
  const [isDropdownOpen, setIsDropdownOpen] = useState(isOpen)
  useEffect(() => setIsDropdownOpen(isOpen), [isOpen]);
  useEffect(() => isDropdownOpen ? onOpen : onClose, [isDropdownOpen]);

  const handleOpen = () => {
    setIsDropdownOpen(!isDropdownOpen);
    isDropdownOpen ? onOpen : onClose;
  }

  return (
    <div className={styles.container}>
      <div className={styles.btn} onClick={() => { handleOpen() }}>
        {button}
      </div>
      <div className={styles.karma}>
        {karma}
      </div>
      {isDropdownOpen && (
        <div className={styles.listContainer}>
          <div className={styles.list} onClick={() => { setIsDropdownOpen(false) }}>
            {children}
          </div>
        </div>
      )}
    </div>
  );
}
