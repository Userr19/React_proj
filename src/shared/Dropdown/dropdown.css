.container {
    min-width: 160px;
    position: relative;
    /* flex: 1 1 0%; */
}

.listContainer {
    position: relative;
}

.list {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100% ;
}

.btn {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    right: 50%;
    top: 10px;
    width: 31px;
    height: 31px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.btn:hover {
    background-color: rgba(51, 51, 51, 0.1);
}

.karma {
    position: absolute;
    top: 0;
    right: 14px;
}