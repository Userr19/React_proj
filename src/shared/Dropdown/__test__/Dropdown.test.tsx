/** 
    @jest-environment-jsdom
*/

import { Dropdown } from "../Dropdown"
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

configure({adapter: new Adapter()});

describe('Dropdown', () => {
    test('Should render', () => {
        const wrapper = shallow(<Dropdown button={<button/>} karma={<div />} children={<div />} isOpen={false}/>);
        expect(wrapper).toBeDefined();
        expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy()
    });

    test('Should render (snapshot)', () => {
        const wrapper = shallow(<Dropdown button={<button/>} karma={<div />} children={<div />} isOpen={false}/>)
        expect(wrapper).toMatchSnapshot()
    })
})