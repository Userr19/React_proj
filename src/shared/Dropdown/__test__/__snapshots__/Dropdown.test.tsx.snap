// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown [Should render (snapshot) 1`] = `ShallowWrapper {}`;

exports[`Dropdown Should render (snapshot) 1`] = `
<div
  className="container"
>
  <div
    className="btn"
    onClick={[Function]}
  >
    <button />
  </div>
  <div
    className="karma"
  >
    <div />
  </div>
</div>
`;
